data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -163,10 +167,19 @@ public static List<DataFile> listPartition(
 
       Path partitionDir = new Path(partitionUri);
       FileSystem fs = partitionDir.getFileSystem(conf);
-      List<FileStatus> fileStatus =
-          Arrays.stream(fs.listStatus(partitionDir, HIDDEN_PATH_FILTER))
-              .filter(FileStatus::isFile)
-              .collect(Collectors.toList());
+      List<FileStatus> fileStatus;
+      if (fs.exists(partitionDir)) {
```

Comment:
I'm actually a little worried now that we are going to silently error out on a permissions issue here. Should we have this behavior behind a flag?

Reply:
We added a flag "checkDuplicateFiles" to Spark Table util.

I'm afraid we don't have a clean way of adding more parameters to this method method but i'm still not confident  we should go from failure to warning for everyone all at once.

Reply:
That makes sense. Should we add the flag to the Spark migrate and snapshot procedures as well?

Reply:
That makes sense to me, but you can do that in a followup if you don't want this PR to get too large